File: backjoon/1260 DFS와 BFS.cs
using System;
using static System.Math;
using System.Collections;
using System.Collections.Generic;

namespace Mainapp
{
    class Program
    {
        static int node;
        static int edge;
        static int start;

        static void dfs(ref List<Tuple<int, int>> graphData, bool[] node_visit, int start)
        {
            if (node_visit[start] == true)
                return;
            node_visit[start] = true;
            Console.Write(start + " ");
            List<int> dest_node = new List<int>();
            for (int i = 0; i < graphData.Count; i++)
            {
                int first = graphData[i].Item1;
                int second = graphData[i].Item2;
                if (first == start)
                {
                    dest_node.Add(second);
                }
                else if (second == start)
                {
                    dest_node.Add(first);
                }
            }
            dest_node.Sort();
            for (int i = 0; i < dest_node.Count; i++)
            {
                dfs(ref graphData, node_visit, dest_node[i]);
            }
        }


        static void bfs(ref List<Tuple<int, int>> graphData, bool[] node_visit, int start)
        {
            Queue<int> bfs_queue = new Queue<int>();
            bfs_queue.Enqueue(start);
            node_visit[start] = true;
            while (bfs_queue.Count != 0)
            {
                int node = bfs_queue.Dequeue();
                Console.Write(node + " ");
                List<int> temp = new List<int>();
                for (int i = 0; i < graphData.Count; i++)
                {
                    int first = graphData[i].Item1;
                    int second = graphData[i].Item2;
                    if (first == node)
                    {
                        if (node_visit[second] == false)
                        {
                            node_visit[second] = true;
                            temp.Add(second);
                        }
                    }
                    else if (second == node)
                    {
                        if (node_visit[first] == false)
                        {
                            node_visit[first] = true;
                            temp.Add(first);
                        }
                    }
                }
                temp.Sort();
                for (int i = 0; i < temp.Count; i++)
                {
                    bfs_queue.Enqueue(temp[i]);
                }
            }
        }


        static void Main(string[] args)
        {
            string temp = (Console.ReadLine());
            string []temps = temp.Split(' ');
            node = int.Parse(temps[0]);
            edge = int.Parse(temps[1]);
            start = int.Parse(temps[2]);
            List<Tuple<int, int> > graphData = new List<Tuple<int, int> >();

            for (int i = 0; i < edge; i++)
            {
                string[] ab = Console.ReadLine().Split(' ');
                Tuple<int, int> temp2 = new Tuple<int, int>(int.Parse(ab[0]), int.Parse(ab[1]));
                graphData.Add(temp2);
            }

            //for(int i = 0; i < graphData.Count; i++)
            //{
            //    Console.WriteLine(graphData[i]);
            //}

            bool [] dfs_visit = new bool[node + 1];
            bool [] bfs_visit = new bool[node + 1];

            //for (int i = 0; i < node + 1; i++)
            //         {
            //	if (dfs_visit[i] == true || bfs_visit[i] == true)
            //		Console.WriteLine("funck");
            //         }
            dfs(ref graphData, dfs_visit, start);
            Console.WriteLine();
            bfs(ref graphData, bfs_visit, start);

            return;
        }
    }
}

File: backjoon/1260 DFS와 BFS.cpp
#include <iostream>
#include <stdio.h>
#include <string.h>
#include <algorithm>
#include <vector>
#include <queue>

using namespace std;

void dfs(vector<pair<int, int> > &graphData, bool *node_visit, int start)
{
	if (node_visit[start] == true)
		return;
	node_visit[start] = true;
	cout << start << " ";
	vector<int> dest_node;
	for (int i = 0; i < graphData.size(); i++)
	{
		int first = graphData[i].first;
		int second = graphData[i].second;
		if (first == start)
		{
			dest_node.push_back(second);
		}
		else if (second == start)
		{
			dest_node.push_back(first);
		}
	}
	sort(dest_node.begin(), dest_node.end());
	for (int i = 0; i < dest_node.size(); i++)
	{
		dfs(graphData, node_visit, dest_node[i]);
	}
}


void bfs(vector<pair<int, int> >& graphData, bool* node_visit, int start)
{
	queue<int> bfs_queue;
	bfs_queue.push(start);
	node_visit[start] = true;
	while (bfs_queue.size() != 0)
	{
		int node = bfs_queue.front();
		bfs_queue.pop();
		cout << node << " ";
		vector<int> temp;
		for (int i = 0; i < graphData.size(); i++)
		{
			int first = graphData[i].first;
			int second = graphData[i].second;
			if (first == node)
			{
				if (node_visit[second] == false)
				{
					node_visit[second] = true;
					temp.push_back(second);
				}
			}
			else if (second == node)
			{
				if (node_visit[first] == false)
				{
					node_visit[first] = true;
					temp.push_back(first);
				}
			}
		}
		sort(temp.begin(), temp.end());
		for (int i = 0; i < temp.size(); i++)
		{
			bfs_queue.push(temp[i]);
		}
	}
}

int node; int edge; int start;
int main()
{
	scanf_s("%d", &node);
	scanf_s("%d", &edge);
	scanf_s("%d", &start);
	bool* node_visit = new bool[node + 1];
	vector<pair<int, int> >graphData;
	for (int i = 0; i < edge; i++)
	{
		int a, b;
		scanf_s("%d", &a);
		scanf_s("%d", &b);
		graphData.push_back(std::make_pair(a, b));
	}

	dfs(graphData, node_visit, start);
	cout << "\n";
	memset(node_visit, 0, node + 1);
	bfs(graphData, node_visit, start);

	delete[]node_visit;
	return (0);
}
